### PR TITLE
bb_travis - add different versions and operating systems and pictures!

### PR DESCRIPTION
**Pull request by Boruch-Baum**
ref: https://travis-ci.com/Boruch-Baum/emacs-w3m/builds/101443871

This github display is confusing to me; this pull request should only be the
single commit f46b24a 

edit: And one commit to resolve conflicts.

### CODE REVIEW

**Comment by tsuchm** [comment]
The first problem of this PR is that the travis recipe does not install FLIM packge, thus it is impossible to detect byte-compile errors of shimbun modules.

The second problem which is shared by the original travis recipe is that docker container migration is recommended by Travis.

I think the first problem is merge blocker of this PR, and the second problem is optional.

File: ChangeLog [hunk 1]
```diff
@@ -32,11 +40,6 @@
 	* w3m.el (w3m-resize-image-interactive): Refactor function to remove
 	code duplications.
 
-2019-02-12  Boruch Baum  <boruch_baum@gmx.com>
```
Comment (tsuchm) [inline ChangeLog]: Your PR is compromised by different ChangeLog editings.

**Comment by Boruch-Baum** [comment]
> The first problem of this PR is that the travis recipe does not install FLIM packge, thus it is impossible to detect byte-compile errors of shimbun modules.

My first guess is that the way to handle that is to add to the travis `before_script` section, a batch invocation of emacs to perform something like:
```
     (package-list-packages)
     (package-install "flim")
```
Should `apel` and any other packages also be installed?

> The second problem which is shared by the original travis recipe is that docker container migration is recommended by Travis.

Yes! I knew about that, but didn't fin the documentation, and since the documentation for the other method was so prominent, I went that route. But, you are right.

> I think the first problem is merge blocker of this PR, and the second problem is optional.

There will probably be other hiccups as well. For instance, I don't have a clear idea what version of the build tools are being used, and haven't gotten travis to display stdout when I included `foo --version` in the build script.

Also, I see that it seems osx does not install `w3m` by default, so that needs to be done explicitly by travis.

The nature of this work may mean many small experimental commits, so I took to being liberal, and then at the end performing a wholesale 'squash' using `git rebase`. It went smoothly, and is why there was originally only a single commit to this PR.

**Comment by tsuchm** [comment]
Could you explain the reason why you restrict the branches checked by Travis into master and bb-travis?  I would like to check all branches.

**Comment by Boruch-Baum** [comment]
> Could you explain the reason why you restrict the branches checked by Travis into master and bb-travis? I would like to check all branches.

@tsuchm : Yes, but first, I'll remind you that I'm no expert in travis, so I'm not sure that my thoughts are true.

1.  The project has many branches, but seems to only actively commit and take pull requests to the master branch, and the other branches seem just historical. I wanted to avoid needless builds and jobs. It's true that using travis doesn't cost ME anything, but there has got to be someone paying for the computing power, accumulated over time, so I don't want to abuse the kindness.

2. I used my own version of the repository as guidance. There, I have many active branches, but I'm not sure yet that I want travis to perform all those builds every time I push a merge among the branches. For example, my bb_upstream should always be identical to your master, so it doesn't make sense to run travis on merges into that branch. My bb_quibbles is meant to be a persistent development branch, and I have other development branches for issue-specific pull requests, but do they really need travis on every commit push? I thought that they can wait until PR time, when travis will catch them on your side since they are being merged into your master.

3. Maybe on your side, it will make sense to enable all branches, and see how travis behaves in practice.

4. What I would really like is to able to tell travis that if the build request was from my account, to restrict it one way, and if it was from another account, to restrict it a second way. There do exist [travis environment variables](https://docs.travis-ci.com/user/environment-variables/) that might identify the github repo responsible for the build, but I'm not sure how to construct a conditional yaml statement based upon them.

**Comment by Boruch-Baum** [comment]
@tsuchm : I think that I'm at a good stopping point for now, and that the PR can be merged. Travis now processes nine combinations of operating system versions and emacs flavors, pulling in apel/flim, and generating elisp-byte-code. It succeeds in building on linux (ubuntu xenial) emacs 25, 26, snapshot(27), and xemacs. For osx xcode 10.1, it succeeds for emacs 26.

Some of the osx builds have outstanding problems:
1) for aquamacs, there is an unsolved bug in the travis script
2) for xemacs, the clang compiler assert a type mismatch
3) for emacs 25 and emacs 26, there is a failure related to an autoconf script (ref @yamaoka)
4) for emacs 25, w3m-hack fails silently in mid-operation

In order not to hamper the other day-to-day operations of the project, I've configured travis to be forgiving, and to report success even if there are failures for aquamacs, xemacs, and osx/emacs25. That doesn't mean the issues above should be addressed, just that they won't get in the way of other work.

Other travis future work would be caching and docker containers:

1) caching would speed up travis builds for xemacs/osx, and all builds of apel/flim

2) I've looked further into using travis docker containers, and it seems easy for the project to implement. The benefit would be the ability to use linux'es [ other | newer ] than xenial (2016).

**Pull request merged**